### PR TITLE
first batch of services

### PR DESCRIPTION
**Pull request by norydev**
first batch of goodies:
- Heroku
- Mailing (sendgrid)
- AWS S3

What else do we want?

### CODE REVIEW

**Comment by norydev** [comment]
We should add some front-end goodies:
- Google Maps API
- Google Geocoder API
- Google Address Autocomplete API
- OAUTH to login with Facebook/Twitter/etc
- ?

I suggest someone review + merge this one, and then a JS guru can add some client goodies. And if we have more time, we can add a few one of those:
- Twillio
- Slack
- Stripe (not so relevant in Turkey)
- Airbrake / newrelic / Any CI

@Pisano/developers who wants to review? Who wants to add some Google API goodies?

File: SW305.md [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+
+As Rails has a very big and active community, there is a chance someone already wrote a gem (library) to connect with the API we want to connect to. [The Ruby Toolbox](https://www.ruby-toolbox.com/) is a greate place to start to find these gems.
+
+Let's look at how to implement a few services in our Rails application:
+
+## Hosting and deployment
+
+### PostgreSQL
+
+It's about time we change Database. So far we have been using SQLite3 (you may not have mentioned). We will now use PostgreSQL which integrates well with Rails.
```
Comment (mikesoylu) [inline SW305.md]: _you may not have mentioned_ or _you may not have noticed_?

File: SW305.md [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+  gem 'rails_12factor'
+  gem 'puma'
+end
+```
+
+```sh
+$ bundle install
+```
+
+Then add this line to the `Procfile file:
```
Comment (mikesoylu) [inline SW305.md]: no closing backtick(`)

**Comment by mikesoylu** [comment]
@NoryDev had conflicts with master rebased.

**Comment by norydev** [comment]
@mikesoylu made the corrections and rebase

**Pull request merged**